### DISCUSSION TITLE
New Display Type setting is required but has no default value

### DISCUSSION THREAD

**Discussion by tschopp-spear** [drivardxrm/TimePicker.PCF#36] [General]
@drivardxrm, 
We've been using your TimePicker control for a couple years now and really like it.  We recently upgraded to the latest version 1.0.0.9 and it's causing problems for us.  One of the new configuration settings, "Display Type", is required but has no default value.  After we upgrade the TimePicker control in an environment, the managed solution of our product fails when we try to upgrade it.  The cause is that our solution has forms that have the TimePicker control on them.  The prior version of our solution was setup to use the older version of TimePicker (1.0.0.6), and thus does not have anything set for the Display Type values.  During the upgrade of our solution, it has to delete the prior version, and that fails saying:

Solution "OurSolution" failed to import: Solution Upgrade action failed after import as holding. InnerException is: Property displaytype is required, but the declaration is missing. More Details:Parent Type:SystemForm;SubType:main;Parent Id:258537a5-e9ed-ed11-8849-000d3a5c6435;Associated entity:abc_TableName;Control Description Xml:Unexpected schema;Control Reference Xml:Unexpected schema;Error Message:-2146041806;Additional Information:;.

Would it be possible for you to make a new version that either has the Display Type not required, or have it set to a default value "12hr" or "24hr"?

**Comment by drivardxrm** [comment]
@tschopp-spear , Sorry to hear that.
I'll update the version as soon as possible.

**Comment by drivardxrm** [comment]
@tschopp-spear 
Can you try the latest release, I added a default value to the Display Type parameter.
Hope it fixes your issue.

Tell me if it works

**Reply by tschopp-spear** [reply]
@drivardxrm 
Hi David,
I installed the new version and I now see a default value for Display Type.  Big thank you for that.

On a separate note, have you ever tried using TimePicker in a Power Pages site?  We are using it in one and it's not working correctly.  I'm pretty sure it worked fairly recently, but today we noticed the only option available for hours is "12" and the only option for minutes is "00".

It was happening with version 1.0.0.9 and still happens with 1.0.0.10.  I attached a screenshot of the UI and of the HTML code when I inspect it.  Let me know if you have any ideas for how to fix it.

Thanks,
Todd Schopp

![Capture2](https://github.com/drivardxrm/TimePicker.PCF/assets/72826535/db0615be-a897-4427-9108-ada96d5b47af)
![Capture1](https://github.com/drivardxrm/TimePicker.PCF/assets/72826535/e2d6b575-dae1-4b91-aa3e-9d4968472f81)

**Comment by drivardxrm** [comment]
@tschopp-spear, I'm glad the new version fixed your issue.

Unfortunately, I don't have any portal to test the control.

Only thing I can think of is that I introduced 2 new parameters in the latest version
![image](https://github.com/drivardxrm/TimePicker.PCF/assets/38399134/39fa429a-9c21-48b0-b1ab-04a8c8c4b540)

maybe the default values are not passed correctly in the portal.

**Reply by tschopp-spear** [reply]
That fixed it!  I set them both to 1, and it works in the portal now.  Thank you once again!

**Reply by therealvoorism** [reply]
Thanks for this help, worked for me as well. I then changed the minute step to 15 and only show quarter hours!!! Great feature add!!
